Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.NewExpression.cs

```diff
@@ -71,5 +72,26 @@ public ReadOnlyCollection<MemberInfo> Members
         return default(ReadOnlyCollection<MemberInfo>);
       }
     }
+
+#if NETFRAMEWORK_4_0 || SILVERLIGHT_4_0 || SILVERLIGHT_5_0
+    //
+    // Summary:
+    //     Creates a new expression that is like this one, but using the supplied children.
+    //     If all of the children are the same, it will return this expression.
+    //
+    // Parameters:
+    //   arguments:
+    //     The System.Linq.Expressions.NewExpression.Arguments property of the result.
+    //
+    // Returns:
+    //     This expression if no children are changed or an expression with the updated
+    //     children.
+    [Pure]
+    public NewExpression Update(IEnumerable<Expression> arguments)
+    {
+      Contract.Ensures(Contract.Result<NewExpression>() != null);
```

Comment:
I decided not to do this because it is fairly complicated (and the fact I barely understand the logic doesn't help either). In addition, the precondition would involve a call to constructor.GetParameters().Length which the static checker will unlikely be able to prove.